src/test/resources/yaml/basic-producer.yaml

```diff
@@ -12,6 +12,8 @@ acks: all
 retries: 1000
 maxInFlightRequestsPerConnection: 1
 maxPollBlockTime: 20s
+lingerMs: 0
```

Comment:
The way to get that to work in tests with a Duration is to do:
`requestTimeout: 30s`

Reply:
Thanks for your patience, I pushed the suggested changes despite the build failure to make sure with you as far as suggestions go. The error is thrown in the Kafka class `org.apache.kafka.common.config.ConfigDef` and the code in question is trying to coerce a `long` out of a `Duration` object I think:

```java
                case LONG:
                    if (value instanceof Integer)
                        return ((Integer) value).longValue();
                    if (value instanceof Long)
                        return value;
                    else if (value instanceof String)
                        return Long.parseLong(trimmed);
                    else
                        throw new ConfigException(name, value, "Expected value to be a 64-bit integer (long), but it was a " + value.getClass().getName());
```

Reply:
You could cast it to an int when setting the config

Reply:
Brilliant! That seems to have done the trick. Should have thought to do that myself, thanks again for your patience.